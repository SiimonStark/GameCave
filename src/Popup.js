import React, { Component } from 'react';

export default class Popup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showGenre: false,
      genreData: []
    };
  }
  closePopup = () => {
    this.props.setPopup( false );
  }
  getGenreDescript = () => {
    let allGenres = this.props.mainGenres.filter(ea => {
      return this.props.currentGenres.includes(ea.genre);
    });
    this.setState({genreData: allGenres, showGenre: true})
  }
  render() {
    const styleImg = { backgroundImage: 'url(' + this.props.currentImage + ')', backgroundSize: 'cover',  };
    if(this.state.showGenre) {
      return (
        <div className="popup-overlay">
          <div id="genre" className="popup">
            <button className="close" onClick={this.closePopup}>X</button>
            <button className="showGame" onClick={() => {this.setState({showGenre: false})}}>Show Game</button>
            <div className="scroll">
              {this.state.genreData.map(genre => {
                return (
                  <div className="genre-info">
                    <h3>{genre.genre}</h3>
                    <p>{genre.description}</p>
                  </div>
                ) 
              })}
            </div>
          </div>
        </div>
      )
    }
    return (
      <div className="popup-overlay">
        <div className="popup">
          <button className="close" onClick={this.closePopup}>X</button>
          <section className="popup-img" style={styleImg}></section>
          <article className="popup-text">
            <h3 className="popup-title">{(this.props.currentTitle).toUpperCase()}</h3>
            <h4 className="popup-score">
              <p>MetaCritic</p>
              <p>{this.props.currentScore}</p>
              <p>SCORE</p>
            </h4>
            <h4 className="popup-genre">{this.props.currentGenres.join(", ")}</h4>
            <button className="showGenre" onClick={this.getGenreDescript}>Show Genre Info..</button>
            <h4 className="popup-plat">
              <p>Available On:</p>
              <p>{this.props.currentPlatforms.join(", ")}</p>
            </h4>
            <h4 className="popup-multi">
              <p>Multiplayer:</p>
              <p>{(this.props.currentMulti).toString().toUpperCase()}</p>
            </h4>
          </article>
        </div>
      </div>
    )
  }
}



          
          
              