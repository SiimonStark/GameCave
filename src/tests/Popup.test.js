import React from 'react';
import ReactDOM from 'react-dom';
import Popup from '../Popup';
import { shallow } from 'enzyme';

const statePopupMock = jest.fn();

const mockProps = {
  popup: false,
  currentTitle: 'Skyrim',
  currentScore: 99,
  currentGenres: ['Adventure', 'Action'],
  currentPlatforms: ['PC', 'Switch'],
  currentMulti: false,
  currentImage: 'https://i.ebayimg.com/images/g/ZX4AAOSw1vlUrlTm/s-l300.jpg'
}

const mockGenres = [
  {
    "genre": "Action",
    "description": "The action game is a video game genre that emphasizes physical challenges, including handâ€“eye coordination and reaction-time."
  },
  {
    "genre": "Adventure",
    "description": "An adventure game is a video game in which the player assumes the role of protagonist in an interactive story driven by exploration and puzzle-solving."
  },
  {
    "genre": "Horror",
    "description": "The Horror game genre that focuses on survival of the character as the game tries to frighten players with either horror graphics or scary ambience."
  }
]

const popupStateMock = jest.fn();

describe('Popup', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Popup {...mockProps}
              popupState={popupStateMock}
              mainGenres={mockGenres}
              setPopup={statePopupMock}/>
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual({
      showGenre: false,
      genreData: []
    });
  })

  it('should update genredata when button is clicked', () => {
    expect(wrapper.state('showGenre')).toEqual(false);
    wrapper.find('.showGenre').simulate('click');
    expect(wrapper.state('showGenre')).toEqual(true);
  })

  it('should invoke setPopup on button click', () => {
    wrapper.find('.close').simulate('click');
    expect(statePopupMock).toBeCalled();
  })
})