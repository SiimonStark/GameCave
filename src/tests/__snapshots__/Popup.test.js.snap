// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup should match snapshot 1`] = `
<div
  className="popup-overlay"
>
  <div
    className="popup"
  >
    <button
      className="close"
      onClick={[Function]}
    >
      X
    </button>
    <section
      className="popup-img"
      style={
        Object {
          "backgroundImage": "url(https://i.ebayimg.com/images/g/ZX4AAOSw1vlUrlTm/s-l300.jpg)",
          "backgroundSize": "cover",
        }
      }
    />
    <article
      className="popup-text"
    >
      <h3
        className="popup-title"
      >
        SKYRIM
      </h3>
      <h4
        className="popup-score"
      >
        <p>
          MetaCritic
        </p>
        <p>
          99
        </p>
        <p>
          SCORE
        </p>
      </h4>
      <h4
        className="popup-genre"
      >
        Adventure, Action
      </h4>
      <button
        className="showGenre"
        onClick={[Function]}
      >
        Show Genre Info..
      </button>
      <h4
        className="popup-plat"
      >
        <p>
          Available On:
        </p>
        <p>
          PC, Switch
        </p>
      </h4>
      <h4
        className="popup-multi"
      >
        <p>
          Multiplayer:
        </p>
        <p>
          FALSE
        </p>
      </h4>
    </article>
  </div>
</div>
`;
